### PR TITLE
feat: Adapt skore to be compliant on `python >=3.9, <3.13`

### PR DESCRIPTION
**Pull request by thomass-dev**
- Remove requirements files
- Remove use of requirements in Makefile and CI
- Add support of python 3.9 and 3.10
- Fix code and tests

### CODE REVIEW

**Comment by tuscland** [comment]
How hard is it to support 3.13?

File: skore/src/skore/cli/cli.py [hunk 1]
```diff
@@ -65,23 +65,18 @@ def cli(args: list[str]):
...
+        )
+    elif parsed_args.subcommand == "create":
+        __create(
+            project_name=parsed_args.project_name,
+            working_dir=parsed_args.working_dir,
+        )
+    elif parsed_args.subcommand == "quickstart":
+        __quickstart()
+
+    parser.print_help()
```
Comment (rouk1) [inline skore/src/skore/cli/cli.py:82]: Will the `print_help` function be called all the time ?

**Review by rouk1 — CHANGES_REQUESTED** [review]

**Comment by thomass-dev** [comment]
> How hard is it to support 3.13?

Vega is not compatible https://github.com/vega/altair/issues/3620 😢 .

File: skore/src/skore/cli/cli.py (hunk 1 above)
Comment (thomass-dev) [reply skore/src/skore/cli/cli.py]: Bien vu.

**Review by rouk1 — APPROVED** [review]
Wonderful job

**Pull request merged**